src/Functions/getHTTPHeader.cpp

```diff
@@ -0,0 +1,103 @@
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <DataTypes/DataTypeString.h>
+#include <Columns/ColumnString.h>
+#include <Interpreters/Context.h>
+#include <Common/CurrentThread.h>
+#include "Disks/DiskType.h"
+#include "Interpreters/Context_fwd.h"
+#include <Core/Field.h>
+#include <Poco/Net/NameValueCollection.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+}
+
+namespace
+{
+
+/** Get the value of parameter in http headers.
+  * If there no such parameter or the method of request is not
+  * http, the function will return empty string.
+  */
+class FunctionGetHTTPHeader : public IFunction, WithContext
```

Comment:
https://github.com/ClickHouse/ClickHouse/pull/56109#issuecomment-1784166153

This comment referred to have unsafe it is. We should be able to disable the function by config and have a config with which we can forbid certain headers to be returned by this function.